File: ServerSubscriptionManager/Models/Invoice.cs
namespace ServerSubscriptionManager.Models
{
    public class Invoice
    {
        public long Id { get; set; }
        public required long UserId { get; set; }
        public required long SubscriptionPeriodId { get; set; }
        public decimal Amount { get; set; } = 0;
        public User? User { get; set; }
        public SubscriptionPeriod? SubscriptionPeriod { get; set; }
    }
}

File: web/src/app/models/invoice.ts
import { SubscriptionPeriod } from "./subscription-period";
import { User } from "./user";

export interface Invoice {
    id: number;
    userId: number;
    subscriptionPeriodId: number;
    amount: number;
    user?: User;
    subscriptionPeriod?: SubscriptionPeriod;
}